airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -116,8 +116,70 @@
 
 
 class ModelToComponentFactory:
-    @staticmethod
-    def create_component(model_type: Type[BaseModel], component_definition: ComponentDefinition, config: Config) -> type:
+    def __init__(self, is_test_read=False):
```

Comment:
what do you think about passing two flags so the connector builder can control the number of pages and the number of slices independently without having to republish the CDK?

Reply:
seconding this

Reply:
This is part of https://github.com/airbytehq/airbyte/pull/21619 where we pass the maximum number of requests we want to allow. This allows us to calculate the flag